src/algebra/periodic.lean

```diff
@@ -47,6 +48,28 @@ lemma periodic.div [has_add α] [has_div β] (hf : periodic f c) (hg : periodic
   periodic (f / g) c :=
 by simp * at *
 
+lemma periodic.const_smul [add_comm_monoid α] [division_ring γ] [module γ α]
```

Comment:
`γ` doesn't need addition and can be `group_with_zero` (or something weaker), right? Same for the ones below. Have you checked whether you can get away with something weaker than `module`?

Reply:
The weak version of this needs `distrib_mul_action_with_zero` which doesn't exist yet.

Reply:
@eric-wieser and I had a long discussion about this and I'm pretty sure the answer is that we can't get away with something weaker. 
Eric, is that correct?

Reply:
Sorry, upon refreshing the page I see that Eric has already replied.

Reply:
Ok, great!